### DISCUSSION TITLE
How can I create an entity?

### DISCUSSION THREAD

**Discussion by hieutv199x** [medusajs/medusa#2472] [Q&A] [answered]
Hi everyone, I'm newbie, I tried to create a new entity follow by this guide [Create an entity](https://docs.medusajs.com/advanced/backend/entities/), when I try to Create a Migration 
![image](https://user-images.githubusercontent.com/31736006/196749736-eff156d0-dddd-4e69-93ab-52c0ae936aa8.png)

```shell
npx typeorm migration:create -n AddPost --dir src/migrations
```
It return file Migration with nothing todo look like:

![image](https://user-images.githubusercontent.com/31736006/196748377-6b29be83-0de7-4a71-a243-e3ae91a8ed1c.png)

Please, give me an idea to help me for passing this step. Thank you!

**Comment by olivermrbl** [comment, marked as answer]
The `create` command from Typeorm generates empty migrations, so this is expected. We recommend you write your own queries within these migrations to ensure you achieve exactly what you are aiming for. 

That being said, you can let Typeorm generate migrations for you by running their `generate` command. Though, it requires additional configuration steps including setting up a DataSource. You can read more about it in their documentation [here](https://typeorm.io/migrations#generating-migrations).

**Reply by SELO77** [reply]
But, when generating the migration file, it raised an error like 'TypeORMError: Entity metadata for Store#default_currency was not found.' 

I recommend to give datasource.js file for default on create medusajs project 👍 

Thanks for reading
